File: src/hello.f90
program hello_world
    ! We need to use the ISO_C_BINDING module to declare the types compatible with C
    use iso_c_binding
    use rust
    use some_module

    implicit none

    type(rustacean_struct) :: s
    s%number = 42
    s%pi = 3.14
    s%e = SomeEnum_A

    print *, "Hello from Fortran!"
    call hello_from_rust()
    print *, "2 + 3 = ", rustacean_sum(2, 3)
    call display_rustacean_struct(s)
    call hello()

end program hello_world

File: src/hello.cpp
/*
This is the C++ equivalent of the file hello.f90
*/


#include <iostream>

#include <rust.hpp>

constexpr int a = 2;
constexpr int b = 3;

int main(int argc, char* argv[])
{
    std::cout << "Hello from C++!" << std::endl;
    hello_from_rust();
    std::cout << a << " + " << b << " = " << rustacean_sum(a, b) << std::endl;
    
    const rustacean_struct s = {
        42,
        3.14,
        SomeEnum::SomeEnum_A,
    };
    display_rustacean_struct(&s);
    return 0;
}